authors/apps/profiles/views.py

```diff
@@ -0,0 +1,69 @@
+from rest_framework import status
+from rest_framework.generics import RetrieveUpdateAPIView
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from django.contrib.auth import get_user_model
+from django.core.exceptions import ObjectDoesNotExist
+from django.http import Http404
+
+from .models import Profile
+from .renderers import ProfileJSONRenderer
+from .serializers import ProfileSerializer
+from .exceptions import ProfileDoesNotExist
+from drf_yasg.utils import swagger_auto_schema
+
+class UpdateGetUserProfileAPIView(RetrieveUpdateAPIView):
+    permission_classes = (IsAuthenticated,)
+    renderer_classes = (ProfileJSONRenderer,)
+    serializer_class = ProfileSerializer
+
+    @swagger_auto_schema(
+        operation_id='Retrieve User Profile',
+        request_body=serializer_class,
+        responses={201: serializer_class(many=False), 400: 'BAD REQUEST'},
+    )
+    def retrieve(self, request, username, *args, **kwargs):
+        # Try to retrieve the requested profile and throw an exception if the
+        # profile could not be found.
+        try:
+            # We use the `select_related` method to avoid making unnecessary
+            # database calls.
+            profile = Profile.objects.select_related('user').get(
+                user__username=username
+            )
+        except Profile.DoesNotExist:
+            raise ProfileDoesNotExist
```

Comment:
then Please correct me if l am wrong, the doesn't exist will check if that profile doesn't exist. then it will through an error message just wondering why don't you provide a description message for this error.

Reply:
@jennizalwango kindly note that `ProfileDoesNoteExist` is an exception handler for this and upon trigger will through a descriptive message.